packages/SwingSet/src/vats/comms/state.js

```diff
@@ -220,27 +220,36 @@ export function makeState(syscall, identifierBase = 0) {
    * Note that this should only be called *after* all work for a crank is done,
    * because transient zero refCounts are possible during the middle of a crank.
    */
-  function purgeDeadLocalPromises() {
-    if (enableLocalPromiseGC) {
-      for (const lpid of deadLocalPromises.values()) {
-        const refCount = Number(store.get(`${lpid}.refCount`));
-        assert(
-          refCount === 0,
-          X`promise ${lpid} in deadLocalPromises with non-zero refcount`,
-        );
-        let idx = 0;
-        const slots = commaSplit(store.get(`${lpid}.data.slots`));
-        for (const slot of slots) {
-          // Note: the following decrement can result in an addition to the
-          // deadLocalPromises set, which we are in the midst of iterating.
-          // TC39 went to a lot of trouble to ensure that this is kosher.
-          decrementRefCount(slot, `gc|${lpid}|s${idx}`);
-          idx += 1;
+  function processMaybeFree() {
+    // We make a copy of the set, iterate over that, then try again, until
+    // the set is empty. TC39 went to a lot of trouble to make sure you can
+    // add things to a Set while iterating over it, but I think our
+    // `!refCount` check could still be fooled, so it seems safer to use this
+    // approach.
+    while (maybeFree.size) {
+      const lrefs = Array.from(maybeFree.values());
+      lrefs.sort();
+      maybeFree.clear();
+      for (const lref of lrefs) {
+        const { type } = parseLocalSlot(lref);
+        if (type === 'promise' && enableLocalPromiseGC) {
+          const s = store.get(`${lref}.refCount`);
+          if (s) {
+            const refCount = Number(s);
```

Comment:
any particular reason why a `Number` object rather than a `number` primitive?

Reply:
```suggestion
            const refCount = parseInt(s, 10);
```

Reply:
Good point, all I want is a non-negative integer from a string. We're being inconsistent here w.r.t. BigInt vs Number. I'll go with the `parseInt` for now, thanks.